client/state/first-view/actions.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import {
+	FIRST_VIEW_EMBRACE,
+	FIRST_VIEW_DISMISS,
+	FIRST_VIEW_HIDE,
+	FIRST_VIEW_SHOW,
+} from 'state/action-types';
+
+import { FIRST_VIEW_START_DATES } from './constants';
+
+export function init( { user } ) {
+	const userCreated = moment( user.get().date );
+	let eligibleViews = [];
+
+	for ( let view in FIRST_VIEW_START_DATES ) {
+		if ( moment( FIRST_VIEW_START_DATES[ view ] ).isBefore( userCreated ) ) {
+			eligibleViews.push( view );
+		}
+	}
+
+	return {
+		type: FIRST_VIEW_SHOW,
+		views: eligibleViews,
+	};
+}
+
+export function dismissView( { view } ) {
+	return {
+		type: FIRST_VIEW_DISMISS,
+		view,
+	};
+}
+
+export function embraceView( { view } ) {
+	return {
+		type: FIRST_VIEW_EMBRACE,
+		view,
+	};
+}
+
+export function hideViews( { views } ) {
+	return {
+		type: FIRST_VIEW_HIDE,
+		views
+	};
+}
+
+export function showViews( { views, force } ) {
```

Comment:
In what sense are you thinking it doesn't scale? That we will be calling `showView( { view: 'stats' } )` from a bunch of different places?

I was actually just thinking about changing my UI code a bit to move the showing of the first view to a centralized controller callback that all the routes can just include. That would also help solve some of the open issues I have around showing the first view for any section of stats without duplicating code. I will create a separate branch to try this out, in case it ends up not being the direction we want to go.